src/js/stores/AppsStore.js

```diff
@@ -295,7 +314,7 @@ QueueStore.on(QueueEvents.CHANGE, function () {
 AppDispatcher.register(function (action) {
   switch (action.actionType) {
     case AppsEvents.REQUEST_APPS:
-      storeData.apps = processAppsAndEmptyGroups(action.data.body);
+      storeData.apps = processAppsAndGroups(action.data.body);
```

Comment:
I kind of struggle with `processAppsAndGroups` returning an object (`{apps, data}`) when `root == false` and returning an array of `apps` when `root == true`. Seems like the part returning the object is a good candidate for extraction into its own method for a clearer API.